cmd/juju/status/status.go

```diff
@@ -176,23 +192,72 @@ func (c *statusCommand) Init(args []string) error {
 }
 
 var newAPIClientForStatus = func(c *statusCommand) (statusAPI, error) {
-	if c.api != nil {
-		return c.api, nil
+	if c.statusAPI == nil {
+		api, err := c.NewAPIClient()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		c.statusAPI = api
+	}
+	return c.statusAPI, nil
+}
+
+var newAPIClientForStorage = func(c *statusCommand) (storage.StorageListAPI, error) {
+	if c.storageAPI == nil {
+		root, err := c.NewAPIRoot()
+		if err != nil {
+			return nil, err
+		}
+		c.storageAPI = storageapi.NewClient(root)
 	}
-	return c.NewAPIClient()
+	return c.storageAPI, nil
+}
+
+func (c *statusCommand) close() (err error) {
```

Comment:
This isn't right - regardless of where the api came from, we close it. Just use a defer inside Run()